app/assets/javascripts/discourse/app/services/sidebar-state.js

```diff
@@ -86,14 +88,22 @@ export default class SidebarState extends Service {
     const collapsedSidebarSectionKey =
       getCollapsedSidebarSectionKey(sectionKey);
     this.keyValueStore.setItem(collapsedSidebarSectionKey, true);
-    this.collapsedSections.add(collapsedSidebarSectionKey);
+    this.collapsedSections.add(sectionKey);
+    this.appEvents.trigger("sidebar-state:collapse-section", {
```

Comment:
I could live with it... but TBH I'm not a big fan of adding appEvents here. I worry they will be misused (e.g. people might hook onto them, and then modify the DOM).

With https://github.com/discourse/discourse-doc-categories/pull/4 in mind... would it be cleaner/easier to implement "expandActiveSection` and `scrollActiveIntoView` booleans in the existing CustomSidebarPanel API? 🤔